mesa/time.py

```diff
@@ -62,7 +62,7 @@ def add(self, agent: Agent) -> None:
 
         if agent.unique_id in self._agents:
             raise Exception(
-                "Agent with unique id {0} already added to scheduler".format(
+                "Agent with unique id {} already added to scheduler".format(
```

Comment:
Why is this not in f-string?

Reply:
For some reason it wasn't picked up by the automatic conversion. Maybe because it the value uses `repr()` string conversion of the object. I formatted it as a f-string!

Reply:
Note that there are still other places where `.format()` is used:
```
batchrunner.py
27:        return self.MESSAGE.format(self.bad_names)

visualization/UserParam.py
71:        msg = self._ERROR_MESSAGE.format(self.param_type, name)

visualization/modules/PieChartVisualization.py
59:        new_element = new_element.format(fields_json, canvas_width, canvas_height)

visualization/modules/HexGridVisualization.py
71:        new_element = "new CanvasHexModule({}, {}, {}, {})".format(

visualization/modules/BarChartVisualization.py
67:        new_element = new_element.format(

visualization/modules/ChartVisualization.py
72:        new_element = new_element.format(series_json, canvas_width, canvas_height)

visualization/modules/CanvasGridVisualization.py
87:        new_element = "new CanvasModule({}, {}, {}, {})".format(
```

Reply:
The template string ones are not changed for a good reason, but the `new Canvas...` ones should be changed to f-strings.

Reply:
This is out of scope for this PR though.